SQL/0000-00-05-ElectrophysiologyTables.sql

```diff
@@ -201,6 +200,87 @@ CREATE TABLE `physiological_archive` (
     ON DELETE CASCADE
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
+-- SQL tables for BIDS derivative file structure
+-- Create physiological_annotation_file_type table
+CREATE TABLE `physiological_annotation_file_type` (
+    `FileType`        VARCHAR(20)   NOT NULL UNIQUE,
+    `Description` VARCHAR(255),
+    PRIMARY KEY (`FileType`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Create physiological_annotation_file table
+CREATE TABLE `physiological_annotation_file` (
+    `AnnotationFileID`    INT(10)      UNSIGNED NOT NULL AUTO_INCREMENT,
+    `PhysiologicalFileID` INT(10)      UNSIGNED NOT NULL,
+    `FileType`            VARCHAR(20)  NOT NULL,
+    `FilePath`            VARCHAR(255) NOT NULL,
+    PRIMARY KEY (`AnnotationFileID`),
+    CONSTRAINT `FK_phys_file_ID`
+        FOREIGN KEY (`PhysiologicalFileID`)
+        REFERENCES `physiological_file` (`PhysiologicalFileID`),
+    CONSTRAINT `FK_annotation_file_type`
+        FOREIGN KEY (`FileType`)
+        REFERENCES `physiological_annotation_file_type` (`FileType`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Create annotation_archive which will store archives of all the annotation files for
+-- Front-end download
+CREATE TABLE `physiological_annotation_archive` (
+  `AnnotationArchiveID` INT(10) UNSIGNED NOT NULL   AUTO_INCREMENT,
+  `PhysiologicalFileID` INT(10) UNSIGNED NOT NULL,
+  `Blake2bHash`         VARCHAR(128)     NOT NULL,
+  `FilePath`            VARCHAR(255)     NOT NULL,
+  PRIMARY KEY (`AnnotationArchiveID`),
+  CONSTRAINT `FK_physiological_file_ID`
+    FOREIGN KEY (`PhysiologicalFileID`)
+    REFERENCES `physiological_file` (`PhysiologicalFileID`)
+    ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Create annotation_parameter table
+-- Note: This corresponds with the JSON annotation files
+CREATE TABLE `physiological_annotation_parameter` (
+    `AnnotationParameterID` INT(10)      UNSIGNED NOT NULL AUTO_INCREMENT,
+    `AnnotationFileID`      INT(10)      UNSIGNED NOT NULL,
+    `Sources`               VARCHAR(255),
+    `Author`                VARCHAR(50),
+    PRIMARY KEY (`AnnotationParameterID`),
```

Comment:
Do we limit to one entry per file? If so `AnnotationFileID` should be the PK.

Reply:
I believe we do have a limit of one entry per annotation file. This table is keeping track of the `.json` files, which contain the metadata (author etc.) of all the annotations. According to the BIDS specifications, there should only be one of these files.